Add sro2kmm script (#489)
rh-ecosystem-edge/kernel-module-management: tools/sro2kmm/README.md

## Overview

`sro2kmm` is a shell script running an Ansible playbook against an existing OpenShift cluster in order to 
migrate the kernel modules loaded by [Special Resource Operator (SRO)](https://github.com/openshift/special-resource-operator) to the newer [Kernel Module Management operator (KMM)](https://github.com/rh-ecosystem-edge/kernel-module-management).

## Requirements

A running Openshift Cluster with:
- Special Resource Operator installed and managing a DaemonSet which loads a kernel module.

> - Kernel Module Management operator installed and wanted Pods running the same kernel module name as in SRO workload.

Reviewer: Not sure I understand the meaning of "wanted Pods".

A computer with:

- Installed [dialog](https://invisible-island.net/dialog/) package.
  Available in most distributions. `dnf install dialog` will work in Red Hat based distributions.
- Openshift Client (`oc`) with access to OpenShift cluster.
- SSH access to OpenShift cluster nodes.

> - Ansible and Kubernetes python packages installed: `python3.8 -m pip install ansible kubernetes --user`

Reviewer: I think we should rather recommend that people install Ansible with `dnf`.

(Tested on Ansible 2.13.7 and Python 3.8.15.). Further info at [Ansible Documentation](https://docs.ansible.com/ansible/latest/installation_guide/index.html).

## Configuration

The only configuration you should really change is the auth file `tools/sro2kmm/vars/ocp.yaml`

> which includes the credentials for an administrator at your cluster.

Reviewer: ```suggestion
which includes administrator credentials for your cluster.
```

## Usage

Run `./sro2kmm [sr_name] --menu` script to begin the migration.

There are three possible arguments to be used by the script. First is the name of the `SpecialResource` that you intend

>  to migrate, which is a mandatory argument.

Reviewer: ```suggestion
to migrate, which is a mandatory argument.
```

Second argument is related to the settings applied to the kubernetes [k8s_drain](https://docs.ansible.com/ansible/latest/collections/kubernetes/core/k8s_drain_module.html#parameters) module, specifically the delete options used by it.

> Optional delete settings can be set adding argument `--menu` or `m` to the script which will show a checklist for the

Reviewer: ```suggestion
Optional delete settings can be set by providing the `--menu` or `m` arguments to the script, which will show a checklist for the
```

 user to choose:

- `delete_emptydir_data`: continue even if there are pods using emptyDir (local data that will be deleted when the node
 is drained).
- `disable_eviction`: forces drain to use delete rather than evict.
- `force`: continue even if there are pods not managed by a ReplicationController, Job, or DaemonSet.
- `ignore_daemonsets`: ignore DaemonSet-managed pods.

> Third argument to `sro2kmm` is the verbose flag to the ansible playbook. `-v` or `--verbose` will output full 

Reviewer: ```suggestion
Third argument to `sro2kmm` is the verbose flag to the ansible playbook.
`-v` or `--verbose` will output full 
```

ansible verbose.

DaemonSets owned by the specified SpecialResource will be shown in the main selection menu. 
Also a next menu will be shown so you can select containers on which migration patch should

>  be applied.

Reviewer: ```suggestion
be applied.
```
